sycl/test/built-ins/printf.cpp

```diff
@@ -0,0 +1,127 @@
+        // another app
+        // FIXME: strictly check output order once the bug is fixed
+        // CHECK: {{(Hello, World!)?}}
+
+        // Integral types
+        intel::experimental::printf(format_int, 123);
+        intel::experimental::printf(format_int, -123);
+        // CHECK: 123
+        // CHECK-NEXT: -123
+
+        // Floating point types
+        {
+          // You can declare format string in non-global scope, but in this case
+          // static keyword is required
+          static const CONSTANT char format[] = "%f\n";
+          intel::experimental::printf(format, 33.4f);
+          intel::experimental::printf(format, -33.4f);
+        }
+        // CHECK-NEXT: 33.4
+        // CHECK-NEXT: -33.4
+
+        // Vectors
+        cl::sycl::vec<int, 4> v4{5, 6, 7, 8};
+#ifdef __SYCL_DEVICE_ONLY__
+        // On device side, vectors can be printed via native OpenCL types:
+        using ocl_int4 = cl::sycl::vec<int, 4>::vector_t;
+        {
+          static const CONSTANT char format[] = "%v4d\n";
+          intel::experimental::printf(format, (ocl_int4)v4);
+        }
+
+        // However, you are still able to print them by-element:
+        {
+          intel::experimental::printf(format_vec, (int)v4.w(), (int)v4.z(),
+                                      (int)v4.y(), (int)v4.x());
+        }
+#else
+        // On host side you always have to print them by-element:
+        intel::experimental::printf(format_vec, (int)v4.x(), (int)v4.y(),
+                                    (int)v4.z(), (int)v4.w());
+        intel::experimental::printf(format_vec, (int)v4.w(), (int)v4.z(),
+                                    (int)v4.y(), (int)v4.x());
+#endif // __SYCL_DEVICE_ONLY__
+       // CHECK-NEXT: 5,6,7,8
+       // CHECK-NEXT: 8,7,6,5
+
+        // Pointers
+        int a = 5;
+        int *Ptr = &a;
+        // According to OpenCL spec, argument should be a void pointer
+        {
+          static const CONSTANT char format[] = "%p\n";
+          intel::experimental::printf(format, (void *)Ptr);
+        }
+        // CHECK-NEXT: {{(0x)?[0-9a-fA-F]+$}}
+      });
+    });
+    Queue.wait();
+
+    // printf in parallel_for
+    Queue.submit([&](handler &CGH) {
+      CGH.parallel_for<class stream_string>(range<1>(10), [=](id<1> i) {
+        // cast to uint64_t to be sure that we pass 64-bit unsigned value
+        intel::experimental::printf(format_hello_world_2, (uint64_t)i.get(0));
+      });
+    });
+    Queue.wait();
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+    // CHECK-NEXT: {{[0-9]+}}: Hello, World!
+  }
+
+// CHECK: {{(Hello, World!)?}}
```

Comment:
I think you should put another FIXME here as well.

Reply:
Done